ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/validation/AggregateAttestationValidator.java

```diff
@@ -134,7 +138,11 @@ public void addSeenAggregate(final ValidateableAttestation attestation) {
                             CommitteeUtil.get_beacon_committee(
                                 state, aggregateSlot, aggregate.getData().getIndex());
 
-                        final int aggregatorModulo = getAggregatorModulo(beaconCommittee.size());
+                        final int aggregatorModulo =
+                            specProvider
+                                .atSlot(state.getSlot())
```

Comment:
Now here's a fun question - given the state may be an epoch before the attestation, which epoch should we use to find the spec?

I think it's going to be context dependent and probably then have to be reviewed of network upgrades change something where this matters.  In this particular case, to preserve the ability to calculate attestation duties ahead of time, the committees would have to be stable so it would have to be the attestation epoch - 1 (which isn't necessarily the state's epoch here).

I think in this case I'd actually use `attestation.getSlot` so that we're being very consistent about use the spec for the most closely related object (in this case the aggregate we're validating). If we're every stupid enough to change this particular function, we may have to revisit this call or special case the first epoch after the fork change.  Its too hard to predict the exact details of that now so being as predictable as possible is probably the best approach.

Definitely need to think carefully about where we get the epoch from in all these calls...

Reply:
is the 'atSlot' better than using .get(epoch)? It seems like wrapping the epoch calculation makes sense to me, but i'm happy to take feedback...

Reply:
I really like atSlot personally. Saves a lot of boilerplate code.